### DISCUSSION TITLE
PDK should be able to modify modules setup, add features, etc.

### DISCUSSION THREAD

**Discussion by scotje** [puppetlabs/pdk#1025] [Feature Request]
_From @bittner on November 9, 2017 10:1_

Currently, the PDK is designed to generate a module skeleton. IIUC, it is not possible to adjust the module setup (e.g. add supported OS, change software license, update URLs and Puppet version requirement, replace a CI configuration).

I know this may sound stupid or exaggerated (because _"you can simply edit the metadata file"_), but it should be possible to have PDK give us a hand not only for _generating_ an initial setup but also _modify_ the very same parameters later.

Non-trivial UX Feature
------------

This request must be seen from a development speed and error avoidance point of view (_"... simply do the same thing again for updating"_).

Please let me acknowledge that this may not be a trivial task to accomplish.

_Copied from original issue: puppetlabs/pdk#346_

**Comment by scotje** [comment]
_From @DavidS on November 9, 2017 10:21_

Part of this is being planned in https://tickets.puppetlabs.com/browse/PDK-670 (update the module to a new version of the PDK/template). This will include centralising some module configuration in the [`.sync.yml`](https://tickets.puppetlabs.com/browse/PDK-668).

The workflow would then be 

* identify the source of truth for a particular thing (outside the *actual* source of the module, that would be either the `metadata.json`, or the `.sync.yml`)
* change the value in that file
* run `pdk update` to re-apply the template using the changed values
* validate, test, commit result to taste

How does that sound to you?

**Comment by scotje** [comment]
_From @bittner on November 9, 2017 20:23_

I'm not sure whether "outside the actual source of the module" is intuitive.

What I thought of is: If you run the `pdk update module` (or whatever) command it feels like going through the questions of `pdk new module` again, with the current (= then-chosen) values as proposed defaults. So that the workflow would stay the same.

**Comment by scotje** [comment]
_From @bmjen on March 21, 2018 23:51_

JIRA ticket created to track this. https://tickets.puppetlabs.com/browse/PDK-873

**Comment by scotje** [comment]
_From @evidex on June 20, 2018 10:51_

Apologies if this is in the wrong spot, but it would be awesome to be able to do something like `pdk new dependency <forge-module` and have PDK automagically modify the metadata and fixtures.

**Comment by bmjen** [comment]
Initial idea is adding a flag such as `--interview` to `pdk update` to re-run the new module interview and change the metadata.
